client/src/featureform/cli.py

```diff
@@ -52,134 +53,131 @@ def cli():
-            host, options=(('grpc.enable_http_proxy', 0),))
-    elif cert is not None or env_cert_path is not None:
-        if env_cert_path is not None and cert is None:
-            cert = env_cert_path
-        with open(cert, 'rb') as f:
-            credentials = grpc.ssl_channel_credentials(f.read())
-        channel = grpc.secure_channel(host, credentials)
+                "Host value must be set with --host flag or in env as FEATUREFORM_HOST")
+
+    if local:
+        register.state().create_all_local()
     else:
-        credentials = grpc.ssl_channel_credentials()
-        channel = grpc.secure_channel(host, credentials)
-    stub = ff_grpc.ApiStub(channel)
+        channel = tls_check(host, cert, insecure)
+        stub = ff_grpc.ApiStub(channel)
+        register.state().create_all(stub)
+
+    rc = ResourceClient(host)
 
-    if resource_type == "user":
-        GetUser(stub, name)
-    elif resource_type == "feature":
+    if resource_type == "feature":
         if not variant:
-            GetFeature(stub, name)
+            rc.get_feature(name)
         else:
-            GetFeatureVariant(stub, name, variant)
+            rc.get_feature(name, variant)
     elif resource_type == "label":
         if not variant:
-            GetLabel(stub, name)
+            rc.get_label(name)
         else:
-            GetLabelVariant(stub, name, variant)
+            rc.get_label(name, variant)
     elif resource_type == "source":
         if not variant:
-            GetSource(stub, name)
+            rc.get_source(name)
         else:
-            GetSourceVariant(stub, name, variant)
-    elif resource_type == "training-set" or resource_type == "trainingset":
+            rc.get_source(name, variant)
+    elif resource_type == "trainingset" or resource_type == "training-set":
         if not variant:
-            GetTrainingSet(stub, name)
+            rc.get_training_set(name)
         else:
-            GetTrainingSetVariant(stub, name, variant)
-    elif resource_type == "provider":
+            rc.get_training_set(name, variant)
+    elif resource_type == "user":
         if variant:
-            print("Variant not needed.")
-            return
-        else:
-            GetProvider(stub, name)
+            raise ValueError("Variant not needed")
+        rc.get_user(name)
+    elif resource_type == "model":
+        if variant:
+            raise ValueError("Variant not needed")
+        rc.get_model(name)
     elif resource_type == "entity":
         if variant:
-            print("Variant not needed.")
-            return
-        else:
-            GetEntity(stub, name)
-    elif resource_type == "model":
+            raise ValueError("Variant not needed")
+        rc.get_entity(name)
+    elif resource_type == "provider":
         if variant:
-            print("Variant not needed.")
-            return
-        else:
-            GetModel(stub, name)
+            raise ValueError("Variant not needed")
```

Comment:
stylistically I don't like long if/else chains. I feel some sort of dictionary/function of functions/errors would be better suited for this section